src/System.Diagnostics.TraceSource/tests/DefaultTraceListenerClassTests.cs

```diff
@@ -45,6 +60,46 @@ public void Fail()
         }
 
         [Fact]
+        public void Fail_WithoutWriteLineOverride()
+        {
+            var listener = new TestDefaultTraceListener();
+            listener.ShouldOverrideWriteLine = false;
+            listener.Fail("FAIL");
+            listener.ShouldOverrideWriteLine = true;
+            Assert.False(listener.Output.Contains("FAIL"));
+        }
+
+        [Fact]
+        public void Fail_WithLogFile()
+        {
+            var listener = new TestDefaultTraceListener();
+            listener.LogFileName = "LogFile.txt";
+
+            string pathToLogFile = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), listener.LogFileName);
+
+            listener.ShouldOverrideWriteLine = false;
+            listener.Fail("FAIL");
+            listener.ShouldOverrideWriteLine = true;
+            
+            Assert.True(File.Exists(pathToLogFile));
+            Assert.Contains("FAIL", File.ReadAllText(pathToLogFile));
+
+            File.Delete(pathToLogFile);
+        }
+
+        [Fact]
+        public void Fail_LogFileDirectoryNotFound()
+        {
+            // Exception should be handled by DefaultTraceListener.WriteLine so no need to assert.
+            var listener = new TestDefaultTraceListener();
+            listener.LogFileName = @"testfiles\LogFile.txt";
```

Comment:
Nit: consider using a Guid instead of "testfiles", as the latter is a name that could potentially exist at some point, e.g.
```C#
listener.LogFileName = $"{Guid.NewGuid().ToString("N")}\\LogFile.txt";
```